adsmp/app.py

```diff
@@ -261,17 +260,28 @@ def reindex(self, solr_docs, solr_urls, commit=False):
             self.mark_processed([x['bibcode'] for x in solr_docs], type='solr')
         else:
             self.logger.error('%s docs failed indexing', len(errs))
-            # recover from erros by inserting docs one by one
+            # recover from errors by inserting docs one by one
             for doc in solr_docs:
                 try:
+                    self.logger.error('trying individual update_solr %s', doc)
                     solr_updater.update_solr([doc], solr_urls, ignore_errors=False, commit=commit)
                     self.update_processed_timestamp(doc['bibcode'], type='solr')
                     self.logger.debug('%s success', doc['bibcode'])
                 except:
+                    # if individual insert fails, we try once more without fulltext
```

Comment:
i think you should really add the response message (from solr) into the raised exception, and then check for the presence of 'body' in that message - before retrying with the removed fulltext. else we may end up failing twice for every document that has non-body related problem

Reply:
Thanks for the suggestion.  Currently the underlying library (solr_updater.update_solr) does not provide visibility into those details.  Unless you suggest otherwise, I'll change it so more details are provided.  https://github.com/adsabs/ADSMasterPipeline/blob/master/adsmp/solr_updater.py#L238